src/window.rs

```diff
@@ -1346,6 +1347,18 @@ impl Window {
             .maybe_queue_on_main(move |w| w.set_cursor_icon(cursor))
     }
 
+    #[inline]
+    pub fn register_custom_cursor_icon(&self, key: u64, cursor: CustomCursor) {
+        self.window
+            .maybe_queue_on_main(move |w| w.register_custom_cursor_icon(key, cursor))
+    }
+
+    #[inline]
+    pub fn set_custom_cursor_icon(&self, key: u64) {
+        self.window
+            .maybe_queue_on_main(move |w| w.set_custom_cursor_icon(key))
+    }
```

Comment:
I see what you have done here. Definitely didn't think about this problem in our earlier discussion.
It does make sense though that switching between cursors needs to be supported somehow to prevent any unnecessary overhead.

This probably goes back to having to fully build a cursor in `CustomCursor` would alleviate this problem entirely, which is what you had before but had to rework because Web.

I think we should figure out a way without having to resort to IDs. The solution here is probably to revert to `CustomCursor` being fully built by the user so using `set_custom_cursor_icon()` has no overhead. For Web, the solution here is probably to just return an error when we are in the wrong thread (this will only affect old versions of Safari, because other browsers support `OffscreenCanvas` and can do this in any thread).

I apologize for the confusion here, I didn't look at all the changes you had in https://github.com/rust-windowing/winit/pull/3218#discussion_r1389988046 and didn't see that you actually changed the public API.